week1/homework/src/server.js

```diff
@@ -9,7 +9,39 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    const handleRequest = (statusCode, jsonMessage) => {
+      response.writeHead(statusCode, {
+        'Content-Type': 'application/json'
+      });
+      response.end(JSON.stringify(jsonMessage));
+    };
+
+    switch (request.url) {
+
+      case '/state';
+        handleRequest(response);
```

Comment:
Your `handleRequest` function expects 2 arguments but you're only providing one here. Also you're passing the `response` object but this contradicts with your function definition where the arguments you are expecting are the status code and a json message that you send back in your response.